OH-6 Cayuse mod v1.2 support

MetalStormGhost opened the pull request:
Implemented support for OH-6 Cayuse mod version 1.2 by Tobsen & Eightball. Includes support for the Vietnam Asset Pack v1.0 and loadouts by @Starfire13.

![OH-6A](https://github.com/dcs-retribution/dcs-retribution/assets/89945461/c1ed326c-5f1d-45e1-80d2-79c9d61240e2)
![image](https://github.com/dcs-retribution/dcs-retribution/assets/89945461/87f83285-0607-4eba-a67d-03bf8c5502ec)
![image](https://github.com/dcs-retribution/dcs-retribution/assets/89945461/0dd6400d-9c79-444f-ab62-b0c2755c3147)
![image](https://github.com/dcs-retribution/dcs-retribution/assets/89945461/5834ed2d-25d7-4f41-a0b2-e5a13426a2fa)
![image](https://github.com/dcs-retribution/dcs-retribution/assets/89945461/e5261274-2aa4-4271-b204-cc894eb4cecd)

MetalStormGhost committed:
Initial commit of OH-6 Modpack v1.2 + Vietnam Asset Pack v1 support.

MetalStormGhost committed:
Added OH-6 Mod loadouts by @Starfire13

MetalStormGhost committed:
Modified   pydcs_extensions/oh6_vietnamassetpack/oh6_vietnamassetpack.py
Added      resources/units/ground_units/vap_m35_truck.yaml
Added      resources/units/ground_units/vap_mule.yaml
Added      resources/units/ground_units/vap_mutt.yaml
Added      resources/units/ground_units/vap_mutt_gun.yaml
Added      resources/units/ground_units/vap_type63_mlrs.yaml
Added      resources/units/ground_units/vap_vc_bicycle_ak.yaml
Added      resources/units/ground_units/vap_vc_zis.yaml

MetalStormGhost committed:
Modified   changelog.md
Modified   resources/factions/USA 1970 Vietnam War.json
Modified   resources/factions/USA 1971 Vietnam War.json
Modified   resources/units/ground_units/vap_mule.yaml
Added      resources/units/aircraft/OH-6A.yaml

MetalStormGhost committed:
Added the OH-6 to factions and implemented the mod selection in the new game wizard. Added an icon an a banner.

MetalStormGhost committed:
Modified   resources/units/aircraft/OH-6A.yaml

MetalStormGhost committed:
Merge branch 'oh6-mod-v1.2-support' of https://github.com/MetalStormGhost/dcs-retribution into oh6-mod-v1.2-support

MetalStormGhost committed:
Added icons for the Vietnam Asset Pack ground units. Also added an icon for the PT-76 since it was missing. Added a Viet Cong 1970s faction.

MetalStormGhost committed:
Merge remote-tracking branch 'remotes/dcs-retribution/dcs-retribution/dev' into oh6-mod-v1.2-support

MetalStormGhost committed:
Added      resources/units/ships/vap_us_seafloat.yaml

Raffson commented on resources/units/aircraft/OH-6A.yaml:
```diff
@@ -0,0 +1,18 @@
+variants:
+  OH-6A Cayuse: {}
+tasks:
+  BAI: 200
+  CAS: 200
+  OCA/Aircraft: 200
```
Does the Recon task have the ability to use AttackGroup & SeachAndAttack waypoint tasks? Otherwise this might not work...

Raffson replied:
Also, if it actually has these tasks, then we need to add Recon accordingly in `aircraftbehavior.py`

MetalStormGhost replied:
I'm not sure if that is relevant in the case of this mod, since I saw somewhere (a reference which I can't find right now) that the OH-6 might not work on AI hands at all, or might work poorly...

Raffson replied:
Well then we definitely need to add the fallbacks either way because even for player flights I believe it will run into an exception during generation of a CAS/BAI/OCA mission.

Raffson replied:
Just checked in DCS, seems like Reconnaissance doesn't have any waypoint-task to attack, except strafing but that won't help us I'm afraid...

I would replace the tasks with Transport only, although I guess it's somewhat unrealistic because I can't imagine the little bird hauling tanks, but that's about the only auto-generated mission I would allow 😅

At the same time we would _need_ these tasks because otherwise you can't plan those flights manually for players either, but I would opt for something that checks whether it's an OH-6 & it's a player flight, inject the Reconnaissance task for `configure_cas` & `configure_oca_strike`, otherwise don't inject anything and let the exception throw...

Raffson committed:
Merge branch 'dev' into oh6-mod-v1.2-support

Raffson committed:
Adjust tasking for OH-6A

OH-6A is only capable of Transport & Reconnaissance, but we can ignore this if the lead slot is a client. AI however will most likely not support this, but we can add Transport & Air Assault instead...

Raffson committed:
Fix bug in configure task + client override fallback
